packages/backend/src/services/UserAttributesService/UserAttributeUtils.ts

```diff
@@ -28,38 +32,90 @@ export const hasUserAttributes = (
 };
 
 export const exploreHasFilteredAttribute = (explore: Explore) =>
-    Object.values(explore.tables).some((table) =>
-        Object.values(table.dimensions).some(
-            (dimension) => dimension.requiredAttributes !== undefined,
-        ),
+    Object.values(explore.tables).some(
+        (table) =>
+            table.requiredAttributes !== undefined ||
+            Object.values(table.dimensions).some(
+                (dimension) => dimension.requiredAttributes !== undefined,
+            ),
+    );
+
+export const doesExploreMatchRequiredAttributes = (
+    explore: Explore,
+    userAttributes: UserAttributeValueMap,
+) =>
+    explore.tables[explore.baseTable].requiredAttributes === undefined ||
+    hasUserAttributes(
+        explore.tables[explore.baseTable].requiredAttributes,
+        userAttributes,
     );
-export const filterDimensionsFromExplore = (
+
+export const getFilteredExplore = (
```

Comment:
Update logic to:
- throw error if the user doesn't have access to explore.
- remove tables the user doesn't have access to
- remove joins 
- remove fields that depend on tables the user doesn't have access to